### ISSUE TITLE
Type guard with generic `Type[T]`

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1421]
As far as I understand, `assert isinstance(v, ty)` tells Pyright `v` is an instance of `ty`:

```python
def f(obj: Any) -> int:
    assert isinstance(obj, int)
    return obj # OK, because `obj` is inferred as `int`
```

Sometimes I want to use this feature with generics:

```python
def g(klass: Type[T], obj: Any) -> T:
    assert isinstance(obj, klass)
    return obj # ERROR, though `obj` has type `klass`
```

Is it possible for Pyright to infer `obj` is an instance of `klass`?

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in Pyright 1.1.106, which I just published. It will also be included in the next release of Pylance.
